### PR TITLE
Brms 36 matching service

### PR DESCRIPTION
**Pull request by manishsw**
updated with matching service and audit details to store in splunk for firstname, lastname and date of birth.

### CODE REVIEW

File: app/uk/gov/hmrc/brm/services/MatchingService.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import uk.gov.hmrc.brm.utils.MatchingType
+
+
+/**
+  * Created by manish.wadhwani on 28/09/16.
+  */
+trait MatchingService {
+  val CLASS_NAME: String = this.getClass.getCanonicalName
+
+  def performMatch(input: Payload, response: GroResponse, matchingType: MatchingType.Value): ResultMatch = {
```
Comment (chris-ianson) [inline app/uk/gov/hmrc/brm/services/MatchingService.scala:33]: Is it worth having a default for matchingType? If no value is passed it defaults to MatchingType.FULL

Comment (adamconder) [reply app/uk/gov/hmrc/brm/services/MatchingService.scala:33]: @chris-ianson I would agree with that.

File: app/uk/gov/hmrc/brm/services/MatchingAlgorithm.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  def performMatch(payload: Payload, responsePayload: GroResponse): ResultMatch = {
+    val firstNames = firstNamesMatch(Some(payload.firstName), Some(responsePayload.child.firstName))
+    val lastNames = lastNameMatch(Some(payload.lastName), Some(responsePayload.child.lastName))
+    val dates = dobMatch(Some(payload.dateOfBirth), responsePayload.child.dateOfBirth)
+
+    getMatchResult(firstNames, lastNames,dates )
+  }
+
+
+  private def getMatchResult(firstNames : Match, lastNames : Match,  dates: Match) : ResultMatch = {
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/services/MatchingAlgorithm.scala]: @manishsw I'm going to refactor this method and remove it entirely.

File: app/uk/gov/hmrc/brm/services/MatchingService.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    debug(CLASS_NAME, "MatchingType", s"${matchingType}")
+    val algorithm = matchingType match {
+      case MatchingType.FULL => FullMatching
+      case MatchingType.PARTIAL => PartialMatching
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/services/MatchingService.scala:37]: @manishsw I'm going to add a default case for this which is FullMatching

**Review by adamconder — CHANGES_REQUESTED** [review]

**Review by adamconder — APPROVED** [review]

**Pull request merged**